packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx

```diff
@@ -742,14 +745,35 @@ const PublishAction: DocumentActionComponent = ({
         status: 'published',
       });
       if (errors) {
-        toggleNotification({
-          type: 'danger',
-          message: formatMessage({
-            id: 'content-manager.validation.error',
-            defaultMessage:
-              'There are validation errors in your document. Please fix them before saving.',
-          }),
-        });
+        let hasUnreadableRequiredField = false;
+        for (const fieldName of Object.keys(schema.attributes)) {
+          const attribute = schema.attributes[fieldName];
+
+          if (attribute && attribute.required) {
+            const userCanReadThisField = (canReadFields ?? []).includes(fieldName);
+
+            if (!userCanReadThisField) {
+              hasUnreadableRequiredField = true;
+              break;
+            }
+          }
+        }
+
+        if (hasUnreadableRequiredField) {
+          toggleNotification({
+            type: 'danger',
+            message: 'Test error about unreadable required field', // TODO: the new error message ??
```

Comment:
@maccomaccomaccomacco please could you help with the new error message?

Reply:
"Your current permissions prevent access to certain required fields. Please request access from an administrator to proceed."
or
"You don’t have the necessary permissions to fill out all the required fields. Please request access from an administrator to proceed."